internal/juju/client.go

```diff
@@ -182,7 +185,16 @@ func (sc *sharedClient) GetConnection(modelName *string) (api.Connection, error)
 	return conn, nil
 }
 
-func (sc *sharedClient) ModelUUID(modelName string) (string, error) {
+// ModelUUID returns the model uuid for the provided modelIdentifier.
+// The modelIdentifier can be a model name or model uuid. If a name
+// is provided, first search the modelUUIDCache for the uuid. If it's
+// not found, fill the model cache and try again. If the modelIdentifier
+// is a uuid, return that without verification.
+func (sc *sharedClient) ModelUUID(modelIdentifier string) (string, error) {
```

Comment:
Agreed with Kian that this function could do with some refactoring. It would perhaps be good to have a  function called ModelIdentifierToUUID that just converted the name to the UUID via the cache or returned immediately if it already was one, then use this everywhere to do that logic rather than repeating it in separate functions.

Reply:
The first action of this method, is to return the UUID if there is one. 
```
func (sc *sharedClient) ModelUUID(modelIdentifier string) (string, error) {
	if names.IsValidModel(modelIdentifier) {
		return modelIdentifier, nil
	}
```

This method is the one which does what you've suggested.

Reply:
> then use this everywhere to do that logic rather than repeating it in separate functions.

@Aflynn50 I'm unclear what is being referred to here.

Reply:
It was referring to the now removed `ModelName` method and the logic in `ModelType`.

To me, it would make more sense if the `ModelUUID` function was renamed to `ModelIdentifierToUUID`, or `ResolveModelUUID`. To me, the name doesn't make it clear that it returns a UUID and takes an identifier. 

On reflect, what I find weird is that there are multiple functions that take a model identifier. In my opinion, I would have one function that turned a model identifier into a UUID and then others that took a UUID. I.e. imo ModelType should take a UUID and at the call site they should first call `ModelIdentifierToUUID`.

In the end though this is a small thing so whatever you decide is fine by me.